sdk/logs/src/main/java/io/opentelemetry/sdk/logs/export/BatchLogRecordProcessor.java

```diff
@@ -197,6 +199,12 @@ private Worker(
                   "The number of logs processed by the BatchLogRecordProcessor. "
                       + "[dropped=true if they were dropped due to high throughput]")
               .build();
+      logsExportFailureCounter =
+          meter
+              .counterBuilder("logsExportFailure")
```

Comment:
The OTLP exporters already have dedicated metrics to track failures: https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterMetrics.java

Would these serve your needs?

Reply:
@jack-berg Thanks. I think this does address the requirement. I tried finding if something already exists for exporter in general, as this is a generic need for any kind of exporter not just BatchLogExporter.
I enabled 'OTEL_EXPORTER_OTLP_ENDPOINT' and got this output. Let me check with the original reporter of the issue.

```
ScopeMetrics #2
ScopeMetrics SchemaURL:
InstrumentationScope io.opentelemetry.exporters.otlp-grpc
Metric #0
Descriptor:
     -> Name: otlp.exporter.exported
     -> Description:
     -> Unit:
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> success: Bool(false)
     -> type: Str(log)
StartTimestamp: 2024-10-14 10:06:54.40763 +0000 UTC
Timestamp: 2024-10-14 10:10:54.418291 +0000 UTC
Value: 9
Metric #1
Descriptor:
     -> Name: otlp.exporter.seen
     -> Description:
     -> Unit:
     -> DataType: Sum
     -> IsMonotonic: true
     -> AggregationTemporality: Cumulative
NumberDataPoints #0
Data point attributes:
     -> type: Str(log)
StartTimestamp: 2024-10-14 10:06:54.40763 +0000 UTC
Timestamp: 2024-10-14 10:10:54.418291 +0000 UTC
Value: 9
```

Reply:
These metrics should be enabled by default if using [autoconfigure](https://opentelemetry.io/docs/languages/java/configuration/#zero-code-sdk-autoconfigure). Note if not using autoconfigure, you need to carefully order the initialization so that the configured meter provider can be passed to the OTLP exporters for spans and logs to collect internal telemetry.

I'm not sure what `OTEL_EXPORTER_METRICS_ENABLED` is a reference to. Its not a property that's used in this repository.

Reply:
My bad, I didn't backup the entire collector logs and misinterpreted that these metrics need to be enabled. These are present by default.